ast-visual-studio-extension/CxExtension/Services/ASCAService.cs

```diff
@@ -0,0 +1,490 @@
+                            Debug.WriteLine($"Temporary file deleted: {tempFilePath}");
+                        }
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                Debug.WriteLine($"Failed to process document: {ex.Message}");
+            }
+        }
+
+        private void ClearAllMarkers()
+        {
+            foreach (var marker in _activeMarkers)
+            {
+                try
+                {
+                    marker.Invalidate(); // Removes the marker
+                }
+                catch (Exception ex)
+                {
+                    Debug.WriteLine($"Failed to invalidate marker: {ex.Message}");
+                }
+            }
+
+            _activeMarkers.Clear(); // Clear the list to avoid references to deleted markers
+        }
+
+
+        private async Task DisplayDiagnosticsAsync(List<CxAscaDetail> scanDetails, string filePath)
+        {
+            if (scanDetails == null || string.IsNullOrEmpty(filePath)) return;
+
+            await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+
+            var document = _dte.Documents.Cast<Document>()
+                .FirstOrDefault(doc => doc.FullName == filePath);
+
+            if (document == null) return;
+
+            // Create if not exists
+            if (_errorListProvider == null)
+            {
+                _errorListProvider = new ErrorListProvider(ServiceProvider.GlobalProvider);
+            }
+
+            // Clear existing tasks in Error List and remove existing markers
+            _errorListProvider.Tasks.Clear();
+            ClearAllMarkers();
+
+            var textManager = ServiceProvider.GlobalProvider.GetService(typeof(SVsTextManager)) as IVsTextManager2;
+            if (textManager == null) return;
+
+            IVsTextView viewEx = null;
+            IVsTextLines buffer = null;
+
+            try
+            {
+                int hr = textManager.GetActiveView2(1, null, 0, out viewEx);
+                if (ErrorHandler.Failed(hr) || viewEx == null) return;
+
+                hr = viewEx.GetBuffer(out buffer);
+                if (ErrorHandler.Failed(hr) || buffer == null) return;
+
+                WriteToOutputPane($"{scanDetails.Count} security best practice violations were found in {document.FullName}");
+
+
+                foreach (var detail in scanDetails)
+                {
+                    // Add to Error List
+                    var task = new ErrorTask
+                    {
+                        Category = TaskCategory.CodeSense,
+                        ErrorCategory = GetErrorCategory(detail.Severity),
+                        Text = $"{detail.RuleName} - {detail.RemediationAdvise} (ASCA)",
+                        Document = document.FullName,
+                        Line = detail.Line - 1,
+                        Column = 0
+                    };
+
+                    task.Navigate += (s, e) =>
```

Comment:
when we use these parameters?

Reply:
The Navigate event handler expects sender and eventArgs parameters